### PR TITLE
Update the Scheduler to Satisfy Dr. Leyk's Constraints

### PR DESCRIPTION
**Pull request by dantebarbieri**
This PR is largely already deployed because it's mostly backend, but I tweak the label text on the Profile Page as the only Frontend change.

**MERGE PRIORITY**: HIGH

__Changelog:__
* **Scheduling is now based on Weekly Number of Hours.** Before it assigned you to a specific number of sections, and this now gives more control over how many hours one wants to work per week.
* **Scheduling adds a subpriority for higher numbered courses.** Dr. Leyk expressed concerns that 300-level courses (e.g. 313, 312, etc.) are not getting scheduled. As such, we add a subpreference for these courses. As a result when the preferences are being checked, higher course numbers are more preferable and are scheduled first.
* **Scheduling now always represents the Viability View.** Previously the tables were multiqueried during scheduling to pull down information. Now the view is queried which reduces the number of queries significantly. The data is loaded from the view meaning that if a user does not appear in the view, they will not be scheduled.

### CODE REVIEW

**Comment by dantebarbieri** [comment]
Addresses the final change in #97.

File: src/Components/Profile/PreferenceSelector.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ export const PreferenceSelector = () => {
       <div className={`${collapsed? "collapsed " : ""}profile-render`}>
         <div className="hstack">
           <div className="dropdown-label">
-            Preferred Number of Lab Sections:  
+            Preferred Number of Weekly Lab Hours:
           </div>
           <input id={input_id} type="number" placeholder={user.user? `${user.user.desired_number_assignments}` : '2'} style={{margin: '0 5px'}}/>
```
Comment (dantebarbieri) [inline src/Components/Profile/PreferenceSelector.tsx]: Has your PR been merged @benbeauchamp7? Looks like when I checked out `main` this line was reverted somehow. I'm not sure what you did to make it work, but I'll leave this suggestion so you can autocommit if you want.
```suggestion
          <input id={input_id} type="number" value={user.user? `${user.user.desired_number_assignments}` : '2'} style={{margin: '0 5px'}}/>
```

File: src/Components/Profile/PreferenceSelector.tsx [hunk 2]
```diff
@@ -58,7 +58,7 @@ export const PreferenceSelector = () => {
       <div className={`${collapsed? "collapsed " : ""}profile-render`}>
         <div className="hstack">
           <div className="dropdown-label">
-            Preferred Number of Lab Sections:
```
Comment (benbeauchamp7) [inline src/Components/Profile/PreferenceSelector.tsx:61]: Frontend seal of approval

**Review by benbeauchamp7 — APPROVED** [review]
Looks great! Happy to see so many removals

**Review by Shane-McGookey — APPROVED** [review]
Looks good.

**Pull request merged**